### DISCUSSION TITLE
Using FXGL embedded in a JavaFX app

### DISCUSSION THREAD

**Discussion by acisternino** [AlmasB/FXGL#1153] [Q&A] [answered]
Hi,

I am considering using FXGL for a simulation program I am developing.

The idea I have is that the GUI will need/use some standard JavaFX components
to interact with the simulation. Things like a menu bar, a toolbar, a pane
with some controls, etc.

The simulation engine will, in turn, run at full speed and the data it works on
can be used to draw things on screen in the remaining free area of the window
and for this I'd like to use FXGL.

Is there any example or article about using FXGL in such a configuration?

How can I initialize and use FXGL in a regular JavaFX `Application`?

**Comment by AlmasB** [comment, marked as answer]
Hia, here's an [example](https://github.com/AlmasB/FXGL/blob/dev/fxgl-samples/src/main/java/advanced/EmbeddedSample.java)

**Reply by acisternino** [reply]
Awesome, thanks a lot. I'll dive right in.

BTW, I am sorry I completely missed that :)
